### PR TITLE
[BUG][ALERT-SERVER]validate script before alert script

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
## Purpose of the pull request

validate script before alert script

## Brief change log

validate script before alert script
## Verify this pull request

have tested locally.

### CODE REVIEW

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java [hunk 1]
```diff
@@ -54,6 +54,21 @@ private AlertResult executeShellScript(String title, String content) {
             alertResult.setMessage("shell script not support windows os");
             return alertResult;
         }
+        //validate script path in case of injections
+        File shellScriptFile = new File(scriptPath);
+        //validate existence
+        if (!shellScriptFile.exists()) {
+            logger.info("shell script not exist : {}", scriptPath);
```
Comment (caishunfeng) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java]:
```suggestion
            logger.error("shell script not exist : {}", scriptPath);
```

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java [hunk 2]
```diff
@@ -54,6 +54,21 @@ private AlertResult executeShellScript(String title, String content) {
...
+            alertResult.setMessage("shell script not exist : " + scriptPath);
+            return alertResult;
+        }
+        //validate is file
+        if (!shellScriptFile.isFile()) {
+            logger.info("shell script is not a file : {}", scriptPath);
```
Comment (caishunfeng) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java]:
```suggestion
            logger.error("shell script is not a file : {}", scriptPath);
```

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java [hunk 3]
```diff
@@ -54,6 +54,21 @@ private AlertResult executeShellScript(String title, String content) {
             alertResult.setMessage("shell script not support windows os");
             return alertResult;
         }
+        //validate script path in case of injections
+        File shellScriptFile = new File(scriptPath);
```
Comment (caishunfeng) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java:58]: It is better to check whether scriptPath is null.

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java (hunk 1 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java]: done

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java (hunk 2 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java]: done

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java (hunk 3 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java:58]: `path` won't be null is must filled when creating

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**